src/UNL/Knowledge/Records.php

```diff
@@ -124,4 +135,108 @@ protected function getMergedKey($item, $section, $tags)
 
     	return implode(' ', $results);
     }
+
+    protected function getPublicProperties()
+    {
+        $self = $this;
+        $getPublicProperties = function() use ($self) {
+            return get_object_vars($self);
+        };
+        $getPublicProperties = $getPublicProperties->bindTo(null, null);
+
+        return $getPublicProperties();
```

Comment:
And in this (any many other cases), using the core API is about 2x as fast. :rocket: